src/XIVLauncher.Common.Unix/Compatibility/DxvkSettings.cs

```diff
@@ -1,121 +1,132 @@
-                    string conf3 = Path.Combine(home,".config","MangoHud","MangoHud.conf");
-                    if (CheckMangoHudPath(conf1))
-                        mangoHudPath = conf1;
-                    else if (CheckMangoHudPath(conf2))
-                        mangoHudPath = conf2;
-                    else if (CheckMangoHudPath(conf3))
-                        mangoHudPath = conf3;
+                    var home = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+                    var conf1 = Path.Combine(corePath.FullName, "MangoHud.conf");
+                    var conf2 = Path.Combine(home, ".config", "MangoHud", "wine-ffxiv_dx11.conf");
+                    var conf3 = Path.Combine(home, ".config", "MangoHud", "MangoHud.conf");
+                    if (File.Exists(conf1))
+                        mangoHudConfig = new FileInfo(conf1);
+                    else if (File.Exists(conf2))
+                        mangoHudConfig = new FileInfo(conf2);
+                    else if (File.Exists(conf3))
+                        mangoHudConfig = new FileInfo(conf3);
                 }
-                if (CheckMangoHudPath(mangoHudPath))
-                    DxvkVars.Add("MANGOHUD_CONFIGFILE",mangoHudPath);
+
+                if (mangoHudConfig != null && mangoHudConfig.Exists)
+                    DxvkVars.Add("MANGOHUD_CONFIGFILE", mangoHudConfig.FullName);
                 else
-                    DxvkVars.Add("MANGOHUD_CONFIG","");
+                    DxvkVars.Add("MANGOHUD_CONFIG", "");
                 break;
+
             case Dxvk.DxvkHudType.MangoHudFull:
                 DxvkVars.Add("DXVK_HUD","0");
                 DxvkVars.Add("MANGOHUD","1");
                 DxvkVars.Add("MANGOHUD_CONFIG","full");
                 break;
-            // If DxvkHudType is None, or undefined, don't set anything.
-            default:
+
+            case Dxvk.DxvkHudType.None:
                 break;
+
+            default:
+                throw new ArgumentOutOfRangeException();
         }
     }
 
-    public static bool CheckDxvkHudString(string customHud)
+    public static bool CheckDxvkHudString(string? customHud)
     {
-        if (customHud == "1") return true;
         if (string.IsNullOrWhiteSpace(customHud)) return false;
+        if (customHud == "1") return true;
         if (!Regex.IsMatch(customHud,ALLOWED_CHARS)) return false;
+
         string[] hudvars = customHud.Split(",");
-        foreach (var hudvar in hudvars)
-        {
-            if (!Regex.IsMatch(hudvar,ALLOWED_WORDS))
-                return false;
-        }
-        return true;
-    }
 
+        return hudvars.All(hudvar => Regex.IsMatch(hudvar, ALLOWED_WORDS));
+    }
+    
     public static bool CheckMangoHudPath(string mangoPath)
     {
-        return (File.Exists(mangoPath)) ? true : false;
-    }
+        return (File.Exists(mangoPath)) ? true : false;    
```

Comment:
Oops, dropped a bracket.
```suggestion
        return (File.Exists(mangoPath)) ? true : false;
    }
```